src/cli.zig

```diff
@@ -935,18 +934,12 @@ pub const Arguments = struct {
                 ctx.bundler_options.asset_naming = try strings.concat(allocator, &.{ "./", bun.strings.removeLeadingDotSlash(asset_naming) });
             }
 
-            if (comptime FeatureFlags.bake) {
-                if (args.flag("--server-components")) {
-                    if (!bun.FeatureFlags.cli_server_components) {
-                        // TODO: i want to disable this in non-canary
-                        // but i also want to have tests that can run for PRs
-                    }
-                    ctx.bundler_options.server_components = true;
-                    if (opts.target) |target| {
-                        if (!bun.options.Target.from(target).isServerSide()) {
-                            bun.Output.errGeneric("Cannot use client-side --target={s} with --server-components", .{@tagName(target)});
-                            Global.crash();
-                        }
+            if (bun.FeatureFlags.bake and args.flag("--server-components")) {
```

Comment:
no more `comptime`? it's just a few branches but we can easily eliminate them with the enclosing `comptime FeatureFlags.bake`

Reply:
bun.FeatureFlags.bake is a compile time variable. i know this because it is a `const` defined outside of a function. Zig has a semantic guarantee to inline and eliminate this branch, so this is zero branches for the same logic, and presented more concisely. i know this guarantee is always going to hold true because if it did not, many places in the codebase would fail to compile.

additionally, i wrote this code before the version of this was added on main.